From d2ec6fff009ac7c008dbec670ccea83cb1f7066a Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:19 -0400
Subject: [tools] lib/traceevent: Use global QUIET_INSTALL build output

Message-id: <1408953210-26343-192-git-send-email-jolsa@redhat.com>
Patchwork-id: 88855
O-Subject: [PATCH RHEL7.1 BZ1133083 191/322] tools lib traceevent: Use global QUIET_INSTALL build output
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 02a82c7bcd353b8cc4d4f8b3c872b527fbac7c31
Author: Jiri Olsa <jolsa@redhat.com>
Date: Thu Dec 19 14:42:05 2013 +0100

description
===========
Using global QUIET_INSTALL build output variable and factoring plugins
installation so we could have only single install message for plugins:

  INSTALL trace_plugins

Getting rid of local print_install.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/Makefile b/tools/lib/traceevent/Makefile
index 51be8ab..8ea4368 100644
--- a/tools/lib/traceevent/Makefile
+++ b/tools/lib/traceevent/Makefile
@@ -152,10 +152,8 @@ override CFLAGS += $(udis86-flags) -D_GNU_SOURCE
 
 ifeq ($(VERBOSE),1)
   Q =
-  print_install =
 else
   Q = @
-  print_install =  echo '  INSTALL  '$1;
 endif
 
 do_compile_shared_library =   \
@@ -307,22 +305,25 @@ TAGS: force
  --regex='/_PE(\([^,)]*\).*/PEVENT_ERRNO__\1/'
 
 define do_install
- $(print_install)    \
  if [ ! -d '$(DESTDIR_SQ)$2' ]; then  \
   $(INSTALL) -d -m 755 '$(DESTDIR_SQ)$2'; \
  fi;      \
  $(INSTALL) $1 '$(DESTDIR_SQ)$2'
 endef
 
-install_lib: all_cmd install_plugins
- $(Q)$(call do_install,$(LIB_FILE),$(bindir_SQ))
-
-PLUGINS_INSTALL = $(subst .so,.install,$(PLUGINS))
+define do_install_plugins
+ for plugin in $1; do    \
+   $(call do_install,$$plugin,$(plugin_dir_SQ)); \
+ done
+endef
 
-$(PLUGINS_INSTALL): %.install : %.so force
- $(Q)$(call do_install,$<,$(plugin_dir_SQ))
+install_lib: all_cmd install_plugins
+ $(call QUIET_INSTALL, $(LIB_FILE)) \
+  $(call do_install,$(LIB_FILE),$(bindir_SQ))
 
-install_plugins: $(PLUGINS_INSTALL)
+install_plugins: $(PLUGINS)
+ $(call QUIET_INSTALL, trace_plugins) \
+  $(call do_install_plugins, $(PLUGINS))
 
 install: install_lib
 
-- 
1.7.1